### DISCUSSION TITLE
Add support for setting default values for response_model formatting

### DISCUSSION THREAD

**Discussion by dkatzan** [fastapi/fastapi#9217] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

app = FastAPI(response_model_exclude_none=True)

@app.get("/")
def read_root():
    return {"Hello": "World"}
```

### Description

I would like to have the ability for setting default values for response_model formatting in the app \ router level.
instead of having to define it on the route level in every route

this will be useful in cases where some non-default behavior is desired across all routes, as currently the way to achieve that will be to set the desired value in every route (which is also very error prone when creating new routes, as it is often forgotten) 

### Operating System

macOS

### FastAPI Version

0.92.0

### Python Version

3.10

### Additional Context

I have already prepared and opened a PR, and will be happy to get it reviewed and eventually merged
https://github.com/tiangolo/fastapi/pull/6041

**Comment by mr-st0rm** [comment]
you can use `APIRouter `with `default_response_class`

**Reply by dkatzan** [reply]
How will that be possible?
`default_response_class` is used after pydantic has serialized the object to a json using the `exclude_unset` `exclude_defaults` `exclude_none` params
and maybe in theory a class to support the `exclude_none` is possible to develop and use, I think will not be possible to mimic the other 2 params

**Reply by mr-st0rm** [reply]
in that case maybe yes
